cmd/ontap-cvo.go

```diff
@@ -120,6 +120,54 @@ func createS3User(onPremName string, namespaceStr string, client *kubernetes.Cli
 	return true
 }
 
+/*
+This will create the S3 bucket. Requires the bucketName to be hashed, the nasPath and relevant management and svm information
+https://docs.netapp.com/us-en/ontap-restapi/ontap/post-protocols-s3-buckets.html
+*/
+func createS3Bucket(svmInfo svmInfo, mgmInfo managementInfo, bucketName string, nasPath string) bool {
+	hashedName := bucketName + "todo-implement-this"
+	// Create a string that is valid json, as thats the simplest way of working with this request
+	// https://go.dev/play/p/xs_B0l3HsBw
+	jsonString := fmt.Sprintf(
+		`{
+			"comment": "created via the ZONE controller",
+			"name": "%s",
+			"nas_path": "%s",
+			"type": "nas",
+			"policy" : {
+				"statements": [
+					{
+						"effect": "allow",
+						"actions": [
+							"GetObject",
+							"PutObject",
+							"ListBucket",
+							"GetBucketAcl",
+							"GetObjectAcl"
+						],
+						"resources": [
+							"%s",
+							"%s/*"
+						]
+					}
+				]
+			}
+		}`,
+		hashedName, nasPath, hashedName, hashedName)
```

Comment:
Is that right? The resources param is just hashName twice?

Reply:
I took this from Pat's working request to create a bucket, where he used the following. 
`bucketname`
and
`bucketname/*`
It is possible that we may not need to do this, as the [default property value](https://docs.netapp.com/us-en/ontap-restapi/ontap/post-protocols-s3-buckets.html#default-property-values) would be all objects in the bucket but I wanted to stick with something explicit first